x-pack/plugins/infra/public/containers/logs/log_position/log_position_state.ts

```diff
@@ -60,8 +76,23 @@ const useVisibleMidpoint = (middleKey: TimeKeyOrNull, targetPosition: TimeKeyOrN
 };
 
 export const useLogPositionState: () => LogPositionStateParams & LogPositionCallbacks = () => {
+  // Flag to determine if `LogPositionState` has been fully initialized.
+  //
+  // When the page loads, there might be initial state in the URL. We want to
+  // prevent the entries from showing until we have processed that initial
+  // state. That prevents double fetching.
+  const [isInitialized, setInitialized] = useState<boolean>(false);
+  const initialize = useCallback(() => {
+    setInitialized(true);
+  }, [setInitialized]);
+
+  const [timestampsLastUpdate, setTimestampsLastUpdate] = useState<number>(Date.now());
+  const updateTimestamps = useCallback(() => {
+    setTimestampsLastUpdate(Date.now());
+  }, [setTimestampsLastUpdate]);
```

Comment:
This *could* become parte of the `DateRange` so we can update it atomically. Just want to throw the idea out there, it's up to you :wink:

Reply:
After inspecting this code maybe we don't need it after all.

This is used when calculating the `useMemo`ized timestamps.

```ts
  const startTimestamp = useMemo(() => datemathToEpochMillis(dateRange.startDateExpression), [
    dateRange.startDateExpression,
    timestampsLastUpdate,
  ]);
```

But I think it can be simplified like this:

```ts
  const startTimestamp = useMemo(() => datemathToEpochMillis(dateRange.startDateExpression), [dateRange]);
```

The `dateRange` is a new object every time consumers calls `updateDateRange`, even if the range is the same, so every time a consumer calls the function, the timestamps will reevaulate.

Reply:
Yes, that would work since we control the `dateRange` setter and can therefore ensure that this holds true.

Reply:
Forget what I said. It's used in https://github.com/afgomez/kibana/blob/de97635c3e5c4651d649e45c5e217b6e71e86eff/x-pack/plugins/infra/public/containers/logs/log_entries/index.ts#L306-L308